scripts/sct_smooth_spinalcord.py

```diff
@@ -45,58 +47,72 @@ def update(self, param_user):
-    parser.add_option(name='-param',
-                      type_value=[[','], 'str'],
-                      description="Advanced parameters. Assign value with \"=\"; Separate params with \",\"\n"
-                                  "algo_fitting {bspline, polyfit}: Algorithm for curve fitting. For more information, see sct_straighten_spinalcord. Default="+ param_default.algo_fitting + ".\n",
-                      mandatory=False)
-    parser.usage.addSection('MISC')
-    parser.add_option(name="-r",
-                      type_value="multiple_choice",
-                      description='Remove temporary files.',
-                      mandatory=False,
-                      default_value='1',
-                      example=['0', '1'])
-    parser.add_option(name="-v",
-                      type_value='multiple_choice',
-                      description="verbose: 0 = nothing, 1 = classic, 2 = expended",
-                      mandatory=False,
-                      example=['0', '1', '2'],
-                      default_value='1')
+    # Initialize the parser
+    parser = argparse.ArgumentParser(
+        description=("Smooth the spinal cord along its centerline. Steps are:\n"
+                     "  1) Spinal cord is straightened (using centerline),\n"
+                     "  2) a Gaussian kernel is applied in the superior-inferior direction,\n"
+                     "  3) then cord is de-straightened as originally.\n"),
+        formatter_class=SmartFormatter,
+        add_help=None,
+        prog=os.path.basename(__file__).strip(".py")
+    )
+
+    mandatory = parser.add_argument_group("\nMANDATORY ARGUMENTS")
+    mandatory.add_argument(
+        '-i',
+        metavar=Metavar.file,
+        required=True,
+        help="Image to smooth. Example: data.nii.gz"
+    )
+    mandatory.add_argument(
+        '-s',
+        metavar=Metavar.file,
+        required=True,
+        help="Spinal cord centerline or segmentation. Example: data_centerline.nii.gz"
+    )
+
+    optional = parser.add_argument_group("\nOPTIONAL ARGUMENTS")
+    optional.add_argument(
+        "-h",
+        "--help",
+        action="help",
+        help="Show this help message and exit."
+    )
+    optional.add_argument(
+        '-smooth',
+        metavar=Metavar.list,
+        type=list_type(',', float),
+        default=[0, 0, 3],
+        help=("Sigma (standard deviation) of the smoothing Gaussian kernel (in mm). For isotropic smoothing you only "
+              "need to specify a value (e.g. 2). For anisotropic smoothing specify a value for each axis, separated "
+              "with a comma. The order should follow axes Right-Left, Antero-Posterior, Superior-Inferior "
+              "(e.g.: 1,1,3). For no smoothing, set value to 0.")
+    )
+    optional.add_argument(
+        '-param',
+        metavar=Metavar.list,
+        type=list_type(',', str),
+        help=(f"R|Advanced parameters. Assign value with '='; Separate params with ','\n"
```

Comment:
since this is the only "param", maybe we should just create a flag `-algo-fitting`? I know this will break compatibility, but now is a good time to break it 😉